commit 36277c1adf23158f6d106f9808bd2fd549ccf3ee
Author: Jake Hillion <jake@hillion.co.uk>
Date:   Fri May 2 11:31:30 2025 +0100

    ixgbe: add hack to support Cisco FET-10G transceivers

diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c
@@ -1770,3 +1770,8 @@ int ixgbe_identify_sfp_module_generic(struct ixgbe_hw *hw)
 	      hw->phy.sfp_type == ixgbe_sfp_type_1g_bx_core1)) {
+		if (hw->allow_unsupported_sfp) {
+			e_warn(drv, "WARNING: driver hit jake's special path.\n");
+			hw->phy.type = ixgbe_phy_sfp_intel;
+			return 0;
+		}
 		hw->phy.type = ixgbe_phy_sfp_unsupported;
